File: Medium/Java/LinkedList/OddEvenList.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode oddEvenList(ListNode head) {
        if (head == null)
            return head;
        ListNode res = head;
        ListNode lastOdd = res, nextOdd = res.next;
        while (nextOdd != null && nextOdd.next != null) {
            ListNode p = lastOdd.next;
            lastOdd.next = nextOdd.next;
            nextOdd.next = lastOdd.next.next;
            lastOdd.next.next = p;
            lastOdd = lastOdd.next;
            nextOdd = nextOdd.next;
        }
        return res;
    }
}

File: Medium/Python/LinkedList/OddEvenList.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def oddEvenList(self, head):
        """
            :type head: ListNode
            :rtype: ListNode
            """
        if head == None:
            return None
        odd = head
        even = head.next
        while even != None and even.next != None:
            odd.next, even.next.next, even.next = even.next, odd.next, even.next.next
            even = even.next
            odd = odd.next
        return head
